dmipy/core/modeling_framework.py

```diff
@@ -2191,6 +2194,742 @@ def __call__(self, acquisition_scheme, **kwargs):
+          linear and non-linear parameters. MIX first uses a stochastic
+          algorithm to find the non-linear parameters (non-volume fractions),
+          then estimates the volume fractions while fixing the estimates of the
+          non-linear parameters, and then finally refines the solution using
+          a gradient-descent-based algorithm.
+
+        The fitting process can be readily parallelized using the optional
+        "pathos" package. If it is installed then it will automatically use it,
+        but it can be turned off by setting use_parallel_processing=False. The
+        algorithm will automatically use all cores in the machine, unless
+        otherwise specified in number_of_processors.
+
+        Data with multiple TE are normalized in separate segments using the
+        b0-values according that TE.
+
+        Parameters
+        ----------
+        acquisition_scheme : DmipyAcquisitionScheme instance,
+            An acquisition scheme that has been instantiated using dMipy.
+        data : N-dimensional array of size (N_x, N_y, ..., N_dwis),
+            The measured DWI signal attenuation array of either a single voxel
+            or an N-dimensional dataset.
+        mask : (N-1)-dimensional integer/boolean array of size (N_x, N_y, ...),
+            Optional mask of voxels to be included in the optimization.
+        solver : string,
+            TODO: ?
+        Ns : integer,
+            for brute optimization, decised how many steps are sampled for
+            every parameter. TODO: ?
+        maxiter : integer,
+            for MIX optimization, how many iterations are allowed. TODO: ?
+        N_sphere_samples : integer,
+            for brute optimization, how many spherical orientations are sampled
+            for 'mu'. TODO: ?
+        use_parallel_processing : bool,
+            whether or not to use parallel processing using pathos.
+        number_of_processors : integer,
+            number of processors to use for parallel processing. Defaults to
+            the number of processors in the computer according to cpu_count().
+        directions_for_lut : int or N-by-3 array
+            If integer, this corresponds to the number of points on the
+            hemisphere that will be used as directions for the look-up-table.
+            If array, it corresponds to the list of directions that will be
+            used for building the look up table. Default: 64.
+
+        Returns
+        -------
+        FittedMultiCompartmentAMICOModel: class instance that contains fitted
+        parameters. Can be used to recover parameters themselves or other useful
+        functions.
+        """
+        self._check_tissue_model_acquisition_scheme(acquisition_scheme)
+        self._check_model_params_with_acquisition_params(acquisition_scheme)
+        self._check_acquisition_scheme_has_b0s(acquisition_scheme)
+        self._check_if_volume_fractions_are_fixed()
```

Comment:
We can add a "default" option to calculate the orientation using DTI internally if there is only one orientation in the model, but we need to leave the option open to fix them by hand.

This will be especially important when we are using AMICO_x (in crossings)

Reply:
I would keep the two separate in such a way that the user always knows what's going on. I agree that it is a good idea to implement a shortcut function that computes the orientations with DTI and its output matches the input of the AMICO constructor, but I wouldn't set it as a default. It's one further modelling step that the user must be aware of even if he's distracted. Imho the default behaviour should be that we estimate also the orientations.

Reply:
You're right. Just keep it as a manual thing for now to set the orientation.

In my opinion I think it just needs to be the same as in the regular framework where you fit the orientations like

```
model = MultiCompartmentAmicoModel(models)
model.set_fixed_parameter(ori_of_model, voxel_wise_values)
```

and then the model can just be fitted with `model.fit()`